streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -144,6 +149,21 @@ public ProcessorNode build() {
         }
     }
 
+    private static class ProcessorNodeCacheFlushListener implements CacheFlushListener<Object, Object> {
```

Comment:
This is out of the scope of this PR: we are now coupling the flushing on the store with forwarding on the processor node here, which is admittedly not ideal; and partly because of this we are adding this function only in the DSL and it from user-facing APIs. Moving forward we should consider a cleaner mechanism for allowing users to decouple flexible flushing (which is a notion of the store) with forwarding (which is a notion of the processor node).